nexus/src/sagas.rs

```diff
@@ -1289,6 +1289,86 @@ async fn sdc_regions_ensure(
 
     let block_size = datasets_and_regions[0].1.block_size;
 
+    // If requested, back disk by image
+    let osagactx = sagactx.user_data();
+    let params = sagactx.saga_params();
+    let opctx = OpContext::for_saga_action(&sagactx, &params.serialized_authn);
+    let read_only_parent: Option<
+        Box<sled_agent_client::types::VolumeConstructionRequest>,
+    > = if let Some(image_id) = params.create_params.image_id {
+        let log = osagactx.log();
+        warn!(log, "grabbing image {}", image_id);
+
+        // TODO: support project images too
+        let global_image = osagactx
+            .datastore()
+            .global_image_fetch(&opctx, image_id)
+            .await
+            .map_err(|e| {
+                ActionError::action_failed(Error::internal_error(
```

Comment:
Why is this and the later one an internal error instead of directly using `e`?

Reply:
I get

```
the trait `From<omicron_common::api::external::Error>` is not implemented for `ActionError`
```

Reply:
Sorry, I meant `.map_err(ActionError::action_failed)`

Reply:
done in ef02727f

Reply:
Thanks.  There's another one further down I think.